### PR TITLE
[ServiceBus] Docs/Docstring/Code improvements

### PR DESCRIPTION
**Pull request by yunhaoling**

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/README.md [hunk 1]
```diff
@@ -150,7 +150,7 @@ with ServiceBusClient.from_connection_string(connstr) as client:
             # If it is desired to halt receiving early, one can break out of the loop here safely.
 ```
 
-> **NOTE:** Any message received with `mode=PEEK_LOCK` (this is the default, with the alternative RECEIVE_AND_DELETE removing the message from the queue immediately on receipt)
+> **NOTE:** Any message received with `receive_mode=PEEK_LOCK` (this is the default, with the alternative ReceiveAndDelete removing the message from the queue immediately on receipt)
```
Comment (KieranBrantnerMagee) [inline sdk/servicebus/azure-servicebus/README.md:153]:
```suggestion
> **NOTE:** Any message received with `receive_mode=PEEK_LOCK` (this is the default, with the alternative RECEIVE_AND_DELETE removing the message from the queue immediately on receipt)
```

File: sdk/servicebus/azure-servicebus/CHANGELOG.md [hunk 2]
```diff
@@ -34,9 +34,13 @@
...
   - Introduced `SessionCannotBeLockedError` which would be raised when the requested session cannot be locked.
-* `ServiceBusClient.get_queue/topic_sender` and `ServiceBusClient.get_queue/subscription_receiver` will now
+* Introduced new client side validation on certain use cases:
+  - Setting `ServiceBusMessage.partition_key` different than `ServiceBusMessage.session_id` will raise `ValueError`.
+  - `ServiceBusClient.get_queue/topic_sender` and `ServiceBusClient.get_queue/subscription_receiver` will now
 raise `ValueError` if the `queue_name` or `topic_name` does not match the `EntityPath` in the connection string used to construct the `ServiceBusClient`.
-
+  - Settling a message that has been peeked will raise `ValueError`.
+  - Settling a message or renewing a lock on a message received in `RECEIVE_AND_DELETE` receive mode will raise `ValueError`.
+  - Setting `session_id`, `reply_to_session_id`, `message_id` and `partition_key` on `ServiceBusMessage` longer 128 characters will raise `ValueError`.
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/CHANGELOG.md:43]:
```suggestion
  - Setting `session_id`, `reply_to_session_id`, `message_id` and `partition_key` on `ServiceBusMessage` longer than 128 characters will raise `ValueError`.
```

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py [hunk 3]
```diff
@@ -20,7 +20,6 @@
 
 if TYPE_CHECKING:
     from azure.core.credentials import TokenCredential
-    from ._common.auto_lock_renewer import AutoLockRenewer
```
Comment (KieranBrantnerMagee) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:23]: This doesn't cause typechecking issues?

Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:23]: no typechecking issue as AutoLockRenewer is in kwargs, it's the sphinx docstring issue that abs ref fixed.

**Review by KieranBrantnerMagee — APPROVED** [review]

**Pull request merged**